docs/ModuleWidgets.md

```diff
@@ -0,0 +1,39 @@
+## Module Widgets
+
+Since LORIS 23.0, LORIS incorporates a concept of "widgets" which
+modules can use to dynamically display graphical widgets in other
+modules when installed and activated. This is a powerful way to
+extend the capabilities of other modules within a new module.
+
+Widgets are can be used by implementing the `getWidgets` method
+on a `Module` module descriptor class. The first parameter of
+`getWidgets` is a type string, allowing modules to define their
+own widget types in a loosely coupled manner. However since the
+interface is abstract enough to be flexible there must be an implied
+contract between what options and their returned values that are not
+enforced by PHP's typing system.
```

Comment:
The wording of this section is a little confusing and I think the grammar may be off. 

From my understanding of this passage, I would conclude that types are not enforced and therefore the widget must be designed with caution because there is not a lot of validation being done. Is that right?

Reply:
There is an '$options' associative array parameter. The PHP type system won't enforce any of the types for key/values passed in that array. This is intentional, since the keys and their value types vary based on the widget type. The document is intended to document the types of widgets that can be added, what `$options` they come with, and what type of widget they expect to be returned in one centralized place that can be referenced when writing a new module.

I agree the wording is off, I was in a rush to write something before I forget what types of widgets we already have. I'm happy to accept suggestions for how to clarify.

Reply:
@driusan What about:
"Widgets can be used by implementing the `getWidgets` method
on a `Module` module descriptor class. The first parameter of
`getWidgets` is a string that define the type of value(s) the module will pass to widget. Using this type, the widget will be able to format the output accordingly." ?

Reply:
made a suggestion below